consensus/enclave/impl/src/lib.rs

```diff
@@ -989,6 +989,13 @@ fn mint_output<T: Digestible>(
     amount: Amount,
     counter: usize,
 ) -> Result<TxOut> {
+    // Check if the token id makes sense right now
+    if !block_version.masked_token_id_feature_is_supported() && amount.token_id != 0 {
+        return Err(Error::FormBlock(
```

Comment:
This is (arguably) a `BlockVersion` error
```suggestion
        return Err(Error::BlockVersion(
```

Reply:
Okay, I agree -- I think it's good to make `form_block` return `FormBlock` errors though because it helps trace the error when it ultimately bubbles up from untrusted side. So maybe when form_block calls `mint_output` we can map this to `FormBlock`, but `mint_output` itself returns `BlockVersion` error in this case.